lib/intercom/scroll_collection_proxy.rb

```diff
@@ -55,8 +56,8 @@ def each(&block)
 
     private
 
-    def deserialize_response_hash(response_hash)
-      top_level_type = response_hash.delete('type')
+    def entity_key_from_response(response_hash)
+      top_level_type = response_hash['type']
```

Comment:
1. Renamed it as it's not deserializing.
2. There's a case in `.each` where we access `.entity_key_from_response` more than once,
So switched accessing type to `response_hash['type']`